chips/nrf52/src/usbd.rs

```diff
@@ -1380,10 +1403,35 @@ impl<'a> Usbd<'a> {
 
         match endpoint {
             0 => {
-                // TODO: the ENDEPOUT0_EP0RCVOUT shortcut could be established instead of manually
-                // triggering the task here.
-                debug_tasks!("- task: ep0rcvout");
-                regs.task_ep0rcvout.write(Task::ENABLE::SET);
+                // We got data on the control endpoint during a CTRL WRITE
+                // transfer. Let the client handle the data, and then finish up
+                // the control write by moving to the status stage.
+
+                // Now we can handle it and pass it to the client to see
+                // what the client returns.
+                self.client.map(|client| {
+                    match client.ctrl_out(endpoint, regs.size_epout[endpoint].get()) {
+                        hil::usb::CtrlOutResult::Ok => {
+                            // We only handle the simple case where we have
+                            // received all of the data we need to.
+                            //
+                            // TODO: Check if the CTRL WRITE is longer
+                            // than the amount of data we have received,
+                            // and receive more data before completing.
+                            self.complete_ctrl_status();
+                        }
+                        hil::usb::CtrlOutResult::Delay => {}
```

Comment:
Note: I'm not sure what would cause a delay here. However, I don't see it happening with only 1 packet of ctrl write data, so until we get a use case with multiple ctrl write transfers, LGTM.